core/root/config/config.go

```diff
@@ -112,6 +112,12 @@ func ConfigFromBytesWithEncryptService(data []byte, encryptService controlplane.
 		return nil, err
 	}
 	c.ProvidedEncryptService = encryptService
+
+	// Uses the same encrypt service for node pools
+	for _, p := range c.NodePools {
+		p.ProvidedEncryptService = encryptService
+	}
```

Comment:
Yes, without this, the `TestMainClusterConfig/WithExperimentalFeatures` test fails to complete the `assertConfig`, probably due to the code that's used to encrypt the bootstrap token for inclusion in `cloud-config-worker`.

I'll take a more careful look in this part to see if I missed something here.